lxd/cluster/raft/file_snapshot.go

```diff
@@ -409,12 +409,18 @@ func (s *FileSnapshotSink) Close() error {
 
 	if runtime.GOOS != "windows" { //skipping fsync for directory entry edits on Windows, only needed for *nix style file systems
 		parentFH, err := os.Open(s.parentDir)
-		defer parentFH.Close()
 		if err != nil {
 			s.logger.Printf("[ERR] snapshot: Failed to open snapshot parent directory %v, error: %v", s.parentDir, err)
 			return err
 		}
 
+		defer func(parentDir string) {
```

Comment:
Just moving the `defer` statement to be after the `if` statement should be enough and would be more in line with what we've done elsewhere.

Reply:
Alright, will update the PR as requested.